File: lint-arwaky/archive/contract/agent_lifecycle_aggregate.py
from abc import ABC, abstractmethod
from pydantic import BaseModel, ConfigDict
from .service_container_aggregate import ServiceContainerAggregate
from ..taxonomy import Duration, ResponseData, AgentStatusVO, BooleanVO

class AgentLifecycleAggregate(BaseModel, ABC):
    """AGGREGATE: Track agent lifecycle state and aggregate system health."""
    model_config = ConfigDict(arbitrary_types_allowed=True)
    container: ServiceContainerAggregate
    status: AgentStatusVO = AgentStatusVO(value="init")
    started: BooleanVO = BooleanVO(value=False)

    @property
    @abstractmethod
    def uptime(self) -> Duration:
        """ARCHITECTURAL COMMITMENT: Uptime tracking."""
        ...

    @abstractmethod
    def mark_started(self) -> None:
        """State transition: started."""
        ...

    @abstractmethod
    async def get_health(self) -> ResponseData:
        """AGGREGATOR: Gather system health data."""
        ...

    @abstractmethod
    def mark_stopped(self) -> None:
        """State transition: stopped."""
        ...

    @abstractmethod
    def mark_degraded(self) -> None:
        """State transition: degraded."""
        ...

File: lint-arwaky/src-rust/contract/agent_lifecycle_aggregate.rs
use async_trait::async_trait;
use std::sync::Arc;
use crate::taxonomy::{Duration, ResponseData, AgentStatusVO, BooleanVO};
use crate::contract::ServiceContainerAggregate;

#[async_trait]
pub trait AgentLifecycleAggregate: Send + Sync {
    /// The orchestration boundary.
    fn container(&self) -> Arc<dyn ServiceContainerAggregate>;
    
    /// Current agent status.
    fn status(&self) -> AgentStatusVO;
    
    /// Whether the agent has started.
    fn started(&self) -> BooleanVO;

    /// ARCHITECTURAL COMMITMENT: Uptime tracking.
    fn uptime(&self) -> Duration;

    /// State transition: started.
    fn mark_started(&self);

    /// AGGREGATOR: Gather system health data.
    async fn get_health(&self) -> ResponseData;

    /// State transition: stopped.
    fn mark_stopped(&self);

    /// State transition: degraded.
    fn mark_degraded(&self);
}
